### PR TITLE
OBPIH-804 Create stock movement from purchase order

### PR DESCRIPTION
**Pull request by DominikaHatala**

### CODE REVIEW

File: grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy [hunk 1]
```diff
@@ -1536,6 +1559,12 @@ class StockMovementService {
         if (shipment && shipment.currentStatus > ShipmentStatusCode.PENDING) {
             shipmentService.rollbackLastEvent(shipment)
             requisitionService.rollbackRequisition(requisition)
+            requisition.requisitionItems.each { requisitionItem ->
+                if (requisitionItem.orderItem) {
+                    OrderItem orderItem = OrderItem.get(requisitionItem.orderItem.id)
+                    orderService.updateOrderItem(orderItem, -requisitionItem.quantity)
```
Comment (awalkowiak) [inline grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy]: What if quantityFulfilled will be lower than requisitionItem.quantity, then we will have value lower than 0

File: src/js/components/form-elements/FieldArrayComponent.jsx [hunk 2]
```diff
@@ -32,6 +32,13 @@ class FieldArrayComponent extends Component {
     const AddButton = fieldsConfig.addButton;
     const { maxTableHeight = 'calc(100vh - 400px)', virtualized } = fieldsConfig;
     const addRow = (row = {}) => fields.push(row);
+    const addRowAfter = (index, row = {}) => {
```
Comment (awalkowiak) [inline src/js/components/form-elements/FieldArrayComponent.jsx]: how about combining these two functions? With default index null, and then do:
> index !== null ? index + 1 : array.length;

File: grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy (hunk 1 above)
Comment (DominikaHatala) [reply grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy]: QuantityFulfilled of order item is a sum of all quantities of requisition items having reference to this order item. During rollback we just reduce quantityFullfiled to 0 by iterating over requisition items.

**Comment by jmiranda** [comment]
I need to create a new branch off develop for all PO-related PRs.

**Comment by jmiranda** [comment]
@DominikaHatala I just created a new branch for the PO feature (OBPIH-803-purchase-order-improvements). Please rebase this PR.

**Comment by DominikaHatala** [comment]
@jmiranda done

**Comment by DominikaHatala** [comment]
@jmiranda Is this ticket is going to be merged in this release?

**Comment by jmiranda** [comment]
@DominikaHatala No, it's going to be merged into a longer running PO feature branch. We'll merge that back to develop once we get a bunch of the PO tickets done.

**Comment by DominikaHatala** [comment]
@jmiranda I have rebased PO feature branch and also this PR so it's ready to be merged.

**Pull request merged**